File: lista de exercicios/questão 9/questao.py
from abc import ABC, abstractmethod

class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def __init__(self, conteudo):
        self.conteudo = conteudo

    def imprimir(self):
        print(f"Imprimindo Relatório: {self.conteudo}")

class Contrato(Imprimivel):
    def __init__(self, conteudo):
        self.conteudo = conteudo

    def imprimir(self):
        print(f"Imprimindo Contrato: {self.conteudo}")

relatorio = Relatorio("Relatório de Vendas")
contrato = Contrato("Contrato de Trabalho")

relatorio.imprimir()
contrato.imprimir()

File: lista de exercicios/questão 9/questao.go
package main

import "fmt"

type Imprimivel interface {
    Imprimir()
}

type Relatorio struct {
    Conteudo string
}

func (r Relatorio) Imprimir() {
    fmt.Printf("Imprimindo Relatório: %s\n", r.Conteudo)
}

type Contrato struct {
    Conteudo string
}

func (c Contrato) Imprimir() {
    fmt.Printf("Imprimindo Contrato: %s\n", c.Conteudo)
}

func main() {
    var documento Imprimivel

    documento = Relatorio{Conteudo: "Relatório de Vendas"}
    documento.Imprimir()

    documento = Contrato{Conteudo: "Contrato de Trabalho"}
    documento.Imprimir()
}
